aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -799,4 +800,241 @@ Tensor sspaddmm(const Tensor& self, const Tensor& mat1, const Tensor& mat2,
   return result;
 }
 
+// --------------------------------------------------------------------
+// sparse.sum()
+// --------------------------------------------------------------------
+Tensor _sparse_sum(const SparseTensor& input) {
+  return input.values().sum();
+}
+
+Tensor _sparse_sum(const SparseTensor& input, ScalarType dtype) {
+  return at::_sparse_sum(input.to(dtype));
```

Comment:
Erm... why not just:

```
return input.values().sum(dtype)
```

Reply:
This would be more efficient too, because you wouldn't have to do a conversion to the correct dtype first, you'd just setup the accumulator correctly (which is what sum does today.)